File: app/src/main/java/b3nac/LayersOfAwesome/WalletContract.java
package b3nac.LayersOfAwesome;

public class WalletContract {

    interface View extends BaseView<WalletContract.Presenter> {

        void showBalance();

        void showWalletAddress();
    }

    interface Presenter extends BasePresenter {

        void getWalletBalance();

        void getWalletAddress();
    }
}

File: app/src/main/java/b3nac/LayersOfAwesome/WalletContract.kt
package b3nac.LayersOfAwesome

class WalletContract {
    internal interface View : BaseView<Presenter?> {
        fun showBalance()
        fun showWalletAddress()
    }

    internal interface Presenter : BasePresenter {
        val walletBalance: Unit
        val walletAddress: Unit
    }
}